### ISSUE TITLE
Better `metric` support for model with multiple outputs in `AutoEstimator` for tf.keras

### ISSUE THREAD

**Issue by yushan111** [intel-analytics/BigDL#3887]
### Problem description 
Currently, user must input the mannually formatted metric name by tf.keras ({layer_name}_{metric_name}) to `metric`, and specify the `metric_mode`. It is not straight-forward enough and the metric format may vary with different tensorflow versions (e.g., "output_acc" for tensorflow==1.15.0, "output_accuracy" for tensorflow==2.4.0)
```python
# in model_creator
model.compile(
        optimizer=optimizer,
        loss={
            'decoder_4': 'mse',
            'output': 'sparse_categorical_crossentropy',
        },
        metrics=['accuracy']
)

# in fit 
autoest.fit(..., metric="output_accuracy", metric_mode="max")
```
### Target 
A more user-friendly interface might be
```python
# in model_creator
model.compile(
        optimizer=optimizer,
        loss={
            'decoder_4': 'mse',
            'output': 'sparse_categorical_crossentropy',
        },
        metrics={
            'decoder_4': 'mse',
            'output': 'accuracy',
        },
)

# in fit
autoest.fit(..., metric={"output": "accuracy"}, metric_mode="max")
```
In that way, the `metric` format is the same as that in `model.compile`, and users don't need to handle the name changes in different tensorflow versions

### Tasks

- [ ] interface change
- [ ] UT tested in tensorflow 1.15
- [ ] Support not passing `metric_mode`

**Comment by jason-dai** [comment]
Does it work for below?

```python
model.compile(loss={"middle": "mse", "output": "mse"},
             optimizer=tf.keras.optimizers.Adam(config["lr"]),
             metrics={"output": "mape"})
...
auto_est.fit(..., 
             metric={"output": "mape"}.
             ...)
```

**Comment by yushan111** [comment]
> Does it work for below?
> 
> ```python
> model.compile(loss={"middle": "mse", "output": "mse"},
>              optimizer=tf.keras.optimizers.Adam(config["lr"]),
>              metrics={"output": "mape"})
> ...
> auto_est.fit(..., 
>              metric={"output": "mape"}.
>              ...)
> ```

We will support it as the target.
